src/Cake.Core/CakeTask.cs

```diff
@@ -35,165 +30,114 @@ public abstract class CakeTask : ICakeTaskInfo
         /// Gets the task's dependencies.
         /// </summary>
         /// <value>The task's dependencies.</value>
-        public IReadOnlyList<CakeTaskDependency> Dependencies => _dependencies;
+        public List<CakeTaskDependency> Dependencies { get; }
 
         /// <summary>
         /// Gets the tasks that the task want to be a dependency of.
         /// </summary>
         /// <value>The tasks that the task want to be a dependency of.</value>
-        public IReadOnlyList<CakeTaskDependency> Dependees => _reverseDependencies;
+        public List<CakeTaskDependency> Dependees { get; }
 
         /// <summary>
         /// Gets the task's criterias.
         /// </summary>
         /// <value>The task's criterias.</value>
-        public IReadOnlyList<Func<ICakeContext, bool>> Criterias => _criterias;
+        public List<CakeTaskCriteria> Criterias { get; }
```

Comment:
`IList`... Or is `ISet` or `ICollection` sufficient?

Reply:
We could change this to an `ICollection` since criterias are inclusive (`AND` based).

Reply:
Changed my mind here, would probably be best to keep compatibility here and use an `IList`.